lib/workers/repository/errors-warnings.ts

```diff
@@ -2,41 +2,41 @@ import type { RenovateConfig } from '../../config/types';
 import { logger } from '../../logger';
 import type { PackageFile } from '../../modules/manager/types';
 import { emojify } from '../../util/emoji';
+import type { DepWarnings } from '../types';
 
 export function getWarnings(config: RenovateConfig): string {
-  if (!config.warnings.length) {
+  if (!config.warnings?.length) {
     return '';
   }
   let warningText = `\n# Warnings (${config.warnings.length})\n\n`;
   warningText += `Please correct - or verify that you can safely ignore - these warnings before you merge this PR.\n\n`;
-  config.warnings.forEach((w) => {
+  for (const w of config.warnings) {
     warningText += `-   \`${w.topic}\`: ${w.message}\n`;
-  });
+  }
   warningText += '\n---\n';
   return warningText;
 }
 
 export function getErrors(config: RenovateConfig): string {
   let errorText = '';
```

Comment:
Maybe also cleaner to move this to the relevant code line and let the edge case have the first line 😉.
Otherwise LGTM